lib/test/t_LeastSquaresExpansion_std.cxx

```diff
@@ -0,0 +1,158 @@
+      const Point expectedCoefficientsHyper = {3.5,1.62542,0,0,0,-0.594723,0,-1.29064,0,0,0,0,0,0,-1.95229,0,0.194929,0,0,0,0,0,0,1.37242,0,0,1.35741,0,-0.0126684,0,0,0,0,0,0,0,0,0,0,0,0,-0.33939,0,0,0,0,0,0,0.409177,0,0.00046142,0,0,0,-1.08975,0,0,0,0,0,0.0459147,0,0,0,0,0,0,0};
+      // Problem parameters
+      UnsignedInteger dimension = 3;
+      Scalar a = 7.0;
+      Scalar b = 0.1;
+      // Create the Ishigami function
+      Description inputVariables(dimension);
+      inputVariables[0] = "xi1";
+      inputVariables[1] = "xi2";
+      inputVariables[2] = "xi3";
+      Description formula(1);
+      formula[0] = (OSS() << "sin(xi1) + (" << a << ") * (sin(xi2)) ^ 2 + (" << b << ") * xi3^4 * sin(xi1)");
+      SymbolicFunction model(inputVariables, formula);
+
+      // Create the input distribution
+      Collection<Distribution> marginals(dimension);
+      marginals[0] = Uniform(-M_PI, M_PI);
+      marginals[1] = Uniform(-M_PI, M_PI);
+      marginals[2] = Uniform(-M_PI, M_PI);
+      ComposedDistribution distribution(marginals);
+
+      // Create the orthogonal basis
+      Collection<OrthogonalUniVariatePolynomialFamily> polynomialCollection(dimension);
+      polynomialCollection[0] = LegendreFactory();
+      polynomialCollection[1] = LegendreFactory();
+      polynomialCollection[2] = LegendreFactory();
+
+      LinearEnumerateFunction enumerateFunction(dimension);
+      OrthogonalProductPolynomialFactory productBasis(polynomialCollection, enumerateFunction);
+
+      // Create the doe
+      UnsignedInteger degree = 10;
+      UnsignedInteger basisSize = enumerateFunction.getBasisSizeFromTotalDegree(degree);
+      UnsignedInteger samplingSize = 8192;
+      UnsignedInteger marginalSize = degree + 5;
+
+      Collection<WeightedExperiment> doeList;
+      RandomGenerator::SetSeed(0);
+      doeList.add(LowDiscrepancyExperiment(LowDiscrepancySequence(SobolSequence()), distribution, samplingSize));
+      doeList.add(GaussProductExperiment(distribution, Indices(distribution.getDimension(), marginalSize)));
+      for (UnsignedInteger i = 0; i < doeList.getSize(); ++i)
+        {
+          // Sampling
+          Point weights;
+          Sample inputSample(doeList[i].generateWithWeights(weights));
+          const Scalar wMin = *std::min_element(weights.begin(), weights.end());
+          const Scalar wMax = *std::max_element(weights.begin(), weights.end());
+          Sample outputSample(model(inputSample));
+          // Check the full constructors
+          Description methodNames = {"SVD", "QR", "Cholesky"};
+          for (UnsignedInteger n = 0; n < 3; ++n)
```

Comment:
Is there a C++ way to compute the number of items in `methodNames`, so that the magic constant 3 disappears from the `for` loop?

Reply:
Ok, I replaced 3 by methodNames.getSize().